src/search/Search.tsx

```diff
@@ -99,30 +97,30 @@ export function Search({ controller, onClose }: SearchProps) {
             onClick={(event) => {
                 const target = event.target as HTMLElement;
                 if (target.id == "search-dialog") { 
-                    _onClose() 
+                    onCloseRequest()
                 }
                 const LI = target.closest("li");
                 if (LI != null) {
                     const index = LI.getAttribute("data-index");
                     if (index) {
                         results[parseInt(index)].action();
-                        setTimeout(_onClose, 1);
+                        setTimeout(onCloseRequest, 1);
                     }
                 }
             }}
```

Comment:
moim zdaniem ten onClick powinien wylecieć bo kilknięcie na ten element domu w tym momencie robi wszystko poza interakcją z tym elementem :D 
1)jeśli chodzi o wywołanie poszczególnych akcji na kilku to rekomenduje zapiąć onClick w resultsView na poszczególnych <li>
2) jeśli chodzi o zamykanie dialogu na click poza okienkiem to polecam najbardziej reużywany snippet  handleClickOutside  https://stackoverflow.com/questions/54391682/detect-click-outside-component-react-hooks :)